src/FastFeatureVector/FeatureVector.scala

```diff
@@ -194,21 +199,39 @@ case class FeatureVector(labelset : Array[String],
         return f
     } */
     def read(iterator: Iterator[String]) {
-        val regex = ("""(.*?)(\+L=("""+labelset.mkString("|")+"""))?[ \t]([^ \t]*)""").r  // .*? is non-greedy
-        // (feature, _, label, value)
-        // matches featurename+L=label 1.0
-        fmap.clear()
-        for (line <- iterator) {
-            val regex(feature, _, label, value) = line
-            if (!fmap.contains(feature)) {
-                fmap(feature) = ValuesMap(0.0, Map())
-            }
-            if (label == null) {
-                fmap(feature).unconjoined = value.toDouble
-            } else {
-                fmap(feature).conjoined(labelToIndex(label)) = value.toDouble // TODO: catch invalid label errors and print labels
-            }
+      val newRegex = ( """(.*?)(\+L=(\:[a-zA-Z0-9-]+))?[ \t]([^ \t]*)""" ).r // .*? is non-greedy
```

Comment:
I'm using IntelliJ and had a custom Code Style selected so that's what's messing up the whitespace. I've set it it 4 space indents now so hopefully it will cause less churn in the future.

What I've done here is replace regex with a simpler (and faster) newRegex which does not hardcode the labelset in it. This has the effect of letting unknown labels through the parse; I then catch these unknown labels and print them all afterwards.